tests/Feature/API/EndpointStreamsTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Tests\Feature\API;
+
+use App\Models\Stream;
+use Illuminate\Foundation\Testing\RefreshDatabase;
+use Illuminate\Support\Carbon;
+use Tests\TestCase;
+
+class EndpointStreamsTest extends TestCase
+{
+    use RefreshDatabase;
+
+    /**
+     * @test
+     */
+    public function it_shows_all_streams(): void
+    {
+        $stream = Stream::factory()->create([
+            'title' => 'Stream #1',
+            'youtube_id' => '1234',
+        ]);
+
+        $response = $this->json(
+            method: 'GET',
+            uri: '/api/v1/streams',
+        );
+
+        $response->assertStatus(
+            status: 200,
+        )->assertHeader(
+            headerName: 'Content-Type',
+            value: 'application/vnd.api+json',
+        )->assertJsonPath(
+            path: '0.id',
+            expect: $stream->id,
+        )->assertJsonPath(
+            path: '0.attributes.identifiers.youtube',
+            expect: $stream->youtube_id,
+        )->assertJsonPath(
+            path: '0.type',
+            expect: 'stream',
+        )->assertJsonPath(
+            path: '0.attributes.title',
+            expect: $stream->title,
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_only_shows_approved_streams(): void
+    {
+        $stream = Stream::factory()->create([
+            'approved_at' => Carbon::now()->subDay(),
+        ]);
+        $stream2 = Stream::factory()->create([
```

Comment:
Can you please use a factory state method here so we can use it like `Stream::factory()->approved()->create();`?

Reply:
OOhh I didn't see that in the code! Of course 😄

Reply:
I am getting this error: `Call to undefined method Database\Factories\StreamFactory::approved()`

Reply:
Sorry, I wasn't clear on that. I wanted you to `add` a new factory method `approved` for streams so that we can use it like that.
https://laravel.com/docs/master/database-testing#factory-states

Reply:
Sorry! Not a problem, leave it with me!